generated/nimodinst/session.py

```diff
@@ -140,22 +140,10 @@ def __len__(self):
         return self._item_count
 
     def __iter__(self):
```

Comment:
This implementation is functionally correct but i think it's not the proper way of doing iterators and goes against the reason why they exist.

This front loads performance and memory hit. I think a more correct implementation would be to have an iterator class which takes a Session and sets its _current_item to 0. __get_item__() constructs the Device. __next__() increments _current_item.

Reply:
If it goes into another class, then we either:
1. open a new session, which is a much larger performance and memory hit
1. pass in the modinst session and somehow keep it open even if the session is explicitly closed.

The only memory and performance is in the Device creation. ModInst has internally already gotten all that information.

Reply:
You would do #2: pass the already open session object. You don't need to keep it open more than you do today... as you have it, if the session is explicitly closed the iterator will stop working because even though the Device instances exist, the calls to the driver will fail.

Reply:
Changed